aloscene/depth.py

```diff
@@ -22,18 +24,55 @@ class Depth(aloscene.tensors.SpatialAugmentedTensor):
     """
 
     @staticmethod
-    def __new__(cls, x, occlusion: Mask = None, *args, names=("C", "H", "W"), **kwargs):
+    def __new__(
+            cls, 
+            x, 
+            occlusion: Mask = None, 
+            is_inverse=False, 
+            scale=None, 
+            shift=None, 
+            *args, 
+            names=("C", "H", "W"), 
+            **kwargs):
+        if is_inverse and not (shift and scale):
+            raise AttributeError('inversed depth requires shift and scale arguments')
         if isinstance(x, str):
             x = load_depth(x)
             names = ("C", "H", "W")
         tensor = super().__new__(cls, x, *args, names=names, **kwargs)
         tensor.add_child("occlusion", occlusion, align_dim=["B", "T"], mergeable=True)
-
+        tensor.add_property('_scale', scale)
+        tensor.add_property('_shift', shift)
+        tensor.add_property('_is_inverse', is_inverse)
         return tensor
 
     def __init__(self, x, *args, **kwargs):
         super().__init__(x)
 
+    def encode_inverse(self, scale: float, shift: float):
+        """Shift and scales the depth values"""
+        depth = self.detach().cpu()
+        if depth._is_inverse:
+            depth = self.encode_absolute()
+        depth.add_property('_scale', scale)
```

Comment:
At this point, the scale property is already added to the augmented tensor.  You only need to set the value you want. depth.scale = scale.

Reply:
`depth = Depth(... is_inverse=True, scale=1, shift=1) # depth already inverted`
`depth = depth.encode_inverse(scale=2, shift=4) # re-invert with new values`

we need to override the scale and shift once we call the encode_inverse method.

Reply:
Yes, my point if that `add_property` is used to create the property in the first place. Once the property is created, you don't need to create the property again. You can simply set the value depth.scale = scale